### PR TITLE
install web3 libs:  viem, wagmi  and connectkit

### PR DESCRIPTION
**Pull request by Diogomartf**
image

**Libs**
- [wagmi](https://wagmi.sh/)
- [viem](https://viem.sh/)
- [@tanstack/react-query](https://tanstack.com/query/v4/docs/framework/react/overview) (wagmi dep)
- [connectkit](https://docs.family.co/connectkit)

### CODE REVIEW

File: app/providers/config.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { ChainId, RPC_LIST } from "@/app/constants";
+
+const chainJsonRpc: Record<number, { http: string }> = {
+  [ChainId.GNOSIS]: {
+    http: RPC_LIST[ChainId.GNOSIS],
+  },
+  [ChainId.ETHEREUM]: {
+    http: RPC_LIST[ChainId.ETHEREUM],
+  },
+};
```
Comment (berteotti) [inline app/providers/config.ts]: We can remove this. It's not being used

File: app/providers/config.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  walletConnectProjectId:
+    process.env.NEXT_PUBLIC_WALLETCONNECT_PROJECT_ID || "",
+  transports: {
+    [mainnet.id]: fallback([http(RPC_LIST[ChainId.ETHEREUM]), http()]),
+    [gnosis.id]: fallback([http(RPC_LIST[ChainId.GNOSIS]), http()]),
+  },
+  appName: "Dapp",
+  appDescription: "Dapp description",
+  appUrl: "https://stackly.app",
+  appIcon: "https://stackly.app/favicon.ico",
```
Comment (berteotti) [inline app/providers/config.ts]: Let's use some generic values.

File: app/providers/providers.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { ConnectKitProvider } from "connectkit";
+
+const queryClient = new QueryClient();
+
+export const Providers = ({ children }: PropsWithChildren) => {
+  return (
+    <WagmiProvider config={config}>
+      <QueryClientProvider client={queryClient}>
+        <ConnectKitProvider
+          options={{ initialChainId: 0 }}
```
Comment (berteotti) [inline app/providers/providers.tsx]: Let's wrap this object in `useMemo`

File: app/components/profile.tsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  const { address } = useAccount();
+  const { data, error, status } = useEnsName({
+    address,
+  });
+  if (status === "pending") return <div>Loading ENS name ...</div>;
+  if (status === "error")
+    return <div>Error fetching ENS name: {error.message}</div>;
+
+  return <div>ENS name: {data}</div>;
+};
```
Comment (berteotti) [inline app/components/profile.tsx:15]: Since this is going to always be deleted. Should we had this directly on `page.tsx` file?

Comment (Diogomartf) [reply app/components/profile.tsx:15]: I think we can have it here, since it keeps the component folder and the example is well organized.

**Pull request merged**